libs/jsonforms-components/src/lib/Controls/Inputs/inputControl.spec.ts

```diff
@@ -135,4 +148,265 @@ describe('input control tests', () => {
+      expect(eventProps.value).not.toMatch(regExNumbers);
+      expect(handleChangeMock.mock.calls.length).toBe(0);
+    });
+
+    it('onBlurForDateControl is valid data', () => {
+      const newProps = { ...props, handleChange: handleChangeMock };
+      onBlurForDateControl({
+        name: 'age',
+        value: '04/04/2024',
+        controlProps: newProps as ControlProps,
+      });
+      expect(newProps.handleChange).toBeCalled();
+    });
+
+    it('onBlurForDateControl is invalid data', () => {
+      const newProps = { ...props };
+      const eventProps = { name: 'age', value: '', controlProps: newProps as ControlProps };
+      onBlurForDateControl(eventProps);
+
+      expect(eventProps.value).toBe('');
+      expect(handleChangeMock.mock.calls.length).toBe(0);
+    });
+
+    it('onBlurTimeControl is valid data', () => {
+      const newProps = { ...props, handleChange: handleChangeMock };
+
+      onBlurForTimeControl({
+        name: 'dateOfEntry',
+        value: '01:01:00 AM',
+        controlProps: newProps as ControlProps,
+      });
+      expect(newProps.handleChange).toBeCalled();
+    });
+
+    it('onChangeForDateControl with data', () => {
+      const newProps = { ...props, handleChange: handleChangeMock };
+
+      onChangeForDateControl({
+        name: 'dateOfEntry',
+        value: '04/04/2024',
+        controlProps: newProps as ControlProps,
+      });
+      expect(newProps.handleChange).toBeCalled();
+    });
+
+    it('onChangeForDateTimeControl with data', () => {
+      const newProps = { ...props, handleChange: handleChangeMock };
+
+      onChangeForDateTimeControl({
+        name: 'dateOfEntry',
+        value: '04/04/2024 01:01:00 AM',
+        controlProps: newProps as ControlProps,
+      });
+      expect(newProps.handleChange).toBeCalled();
+    });
+
+    it('onChangeForNumericControl with data', () => {
+      const newProps = { ...props, handleChange: handleChangeMock };
+
+      onChangeForNumericControl({
+        name: 'age',
+        value: '50',
+        controlProps: newProps as ControlProps,
+      });
+      expect(newProps.handleChange).toBeCalled();
+    });
+    it('onChangeForDateTimeControl with invalid data', () => {
+      const newProps = { ...props };
+
+      const eventProps = {
+        name: 'dateOfEntry',
+        value: '',
+        controlProps: newProps as ControlProps,
+      };
+
+      onChangeForDateTimeControl(eventProps);
+      expect(eventProps.value).toBe('');
+      expect(handleChangeMock.mock.calls.length).toBe(0);
+    });
+
+    it('onChangeForNumericControl with invalid data', () => {
+      const newProps = { ...props };
+
+      const eventProps = {
+        name: 'age',
+        value: 'ab',
+        controlProps: newProps as ControlProps,
+      };
+      onChangeForNumericControl(eventProps);
+      expect(eventProps.value).not.toMatch(regExNumbers);
+      expect(handleChangeMock.mock.calls.length).toBe(0);
+    });
+  });
```

Comment:
A minor suggestion: The test should read like a book e.g.
    it ('can handle invalid data')
reads "it can handle invalid data".  Of course, what is it, in this case.  So that's where descriptions come in handy:
description('tests for onChangeForNumericControl', () => {
   it('can handle invalid data') { .... etc. },
   it('can handle valid data') { ... etc. }